common/isolationgroup/defaultisolationgroupstate/state.go

```diff
@@ -54,14 +54,11 @@ func NewDefaultIsolationGroupStateWatcherWithConfigStoreClient(
 	domainCache cache.DomainCache,
 	cfgStoreClient dynamicconfig.Client, // can be nil, which means global drain is unsupported
 	metricsClient metrics.Client,
+	getIsolationGroups func() []string,
```

Comment:
instead of making it a function, I kinda prefer to define an interface `IsolationGroupProvider` or something with such a method.
But functions can also be easily faked, so it's up to you.

Reply:
I hear your point. I didn't find a convincing reason to do so, just because a drop-in replacement was intended to be so simple. I can't think of any other kind of function to call besides Get, but if you feel strongly I'm happy to change.